dashboard/app/models/plc/course_unit.rb

```diff
@@ -30,4 +30,26 @@ def get_all_possible_learning_resources
   def get_required_learning_module_ids
     plc_learning_modules.required.pluck(:id)
   end
+
+  def get_top_modules_of_each_type_from_user_selections(learning_module_ids)
+    # Preload all of the learning modules, that way we don't have to hit the DB multiple times
+    learning_module_ids.uniq.compact!
```

Comment:
This will not modify the underlying array, because uniq returns a copy. I think you want `learning_module_ids.uniq!.compact!`